calico-vpp-agent/routing/connectivity/vxlan.go

```diff
@@ -53,7 +53,22 @@ func (p *VXLanProvider) configureVXLANNodes() error {
 }
 
 func (p *VXLanProvider) RescanState() {
-	// TODO
+	p.log.Infof("Rescanning existing VXLAN tunnels")
+	p.vxlanIfs = make(map[string]uint32)
+	tunnels, err := p.vpp.ListVXLanTunnels()
+	if err != nil {
+		p.log.Errorf("Error listing VXLan tunnels: %v", err)
+	}
+	nodeIP4 := p.server.GetNodeIP(false)
+	nodeIP6 := p.server.GetNodeIP(true)
+	for _, tunnel := range tunnels {
+		if (tunnel.SrcAddress.Equal(nodeIP4) || tunnel.SrcAddress.Equal(nodeIP6)) &&
+			tunnel.Vni == p.getVXLANVNI() && tunnel.DstPort == p.getVXLANPORT() && tunnel.SrcPort == p.getVXLANPORT() {
```

Comment:
It looks like ListVXLanTunnels does not set SrcPort and DstPort on the tunnel struct, so this check is likely to fail (although it may succeed by chance if getVXLANPort returns 0 😉 ).
Also, I wonder what happens when no port is configured: getVXLANPort will return 0, but will the VPP API return 0 or the default port (4789) ?